python/labbox_ephys/extractors/bandpass_filter.py

```diff
@@ -1,11 +1,24 @@
+import spikeextractors as se
 from .filterrecording import FilterRecording
 import numpy as np
 from scipy import special
+import scipy.fft
+import math
 
 
 class BandpassFilterRecording(FilterRecording):
-    def __init__(self, *, recording, freq_min, freq_max, freq_wid):
-        FilterRecording.__init__(self, recording=recording, chunk_size=3000 * 10)
+    def __init__(self, *, recording: se.RecordingExtractor, freq_min, freq_max, freq_wid):
+        self._padding = 3000
+        target_ram = 100 * 1000 * 1000
+        target_chunk_size = math.ceil(
+            min(
+                recording.get_sampling_frequency() * 30,
+                target_ram / (recording.get_num_channels() * 4)
+            )
+        )
+        # It's important that the fft's have size 2^x. So we prepare the chunk sizes to have size 2^x - 2*padding
+        chunk_size = int(2 ** math.ceil(np.log2(target_chunk_size)) - self._padding * 2)
```

Comment:
FFT is most efficient for chunks of size that is a power of 2

Reply:
There are "edge effects" in the bandpass filter. So the filter is not accurate at the edges of the chunk. Therefore we need some padding, and overlap.